File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionNot.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionNot extends TransformerFunction {
    public TransformerFunctionNot() {
        super(FunctionDescription.of(
            Map.of(
            "style", ArgumentType.of(ArgType.Enum).position(0).defaultEnum("JAVA")
            )
        ));
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var jsStyle = "JS".equals(context.getEnum("style"));
        var adapter = context.getAdapter();
        return !adapter.isTruthy(context.get(null), jsStyle);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionNot.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { isTruthy } from "../JsonHelpers";

class TransformerFunctionNot extends TransformerFunction {
  constructor() {
    super({
      arguments: {
        style: { type: ArgType.Enum, position: 0, defaultEnum: "JAVA" },
      },
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const jsStyle = (await context.getEnum("style")) === "JS";
    return !isTruthy(await context.getUnwrapped(null), jsStyle);
  }
}

export default TransformerFunctionNot;
